apps/mdn/mdn-aws/infra/modules/mdn-updates/main.tf

```
@@ -0,0 +1,188 @@
+locals {
+  aliases     = var.environment == "prod" ? ["updates.developer.mozilla.org", "updates.prod.mdn.mozit.cloud"] : ["updates.developer.allizom.org", "updates.stage.mdn.mozit.cloud"]
+  bucket_name = "updates-${var.environment}-developer-${var.environment == "prod" ? "mozilla" : "allizom"}"
+}
+
+resource "random_id" "rand_var" {
+  keepers = {
+    bucket_name = local.bucket_name
+  }
+
+  byte_length = 8
+}
+
+# IAM user with programmatic access keys that we can use for our GitHub Action secrets
+resource "aws_iam_user" "updates_user" {
+  name = "${local.bucket_name}-${random_id.rand_var.hex}-user"
+  path = "/itsre/"
+
+  tags = {
+    Name        = "${local.bucket_name}-${random_id.rand_var.hex}-user"
+    Environment = var.environment
+    Service     = "MDN"
+    Purpose     = "Cloudfront Updates bucket user"
+  }
+}
+
+# Allow the user full access to the S3 bucket
+resource "aws_iam_user_policy" "updates_user" {
+  name   = "${aws_iam_user.updates_user.name}-iam-policy"
+  user   = aws_iam_user.updates_user.name
+  policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+    {
+        "Effect": "Allow",
+        "Action": "s3:*",
+        "Resource": [
+            "arn:aws:s3:::${local.bucket_name}-${random_id.rand_var.hex}",
+            "arn:aws:s3:::${local.bucket_name}-${random_id.rand_var.hex}/*"
+        ]
+    }]
+}
+EOF
+}
+
+data "aws_iam_policy_document" "public_read" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "s3:GetObject",
+    ]
+
+    principals {
+      type        = "*"
+      identifiers = ["*"]
+    }
+
+    resources = [
+      "arn:aws:s3:::${local.bucket_name}-${random_id.rand_var.hex}/*"
+    ]
+  }
+}
+
+# S3 buckets to be used as origins for the Cloudfront instances
+resource "aws_s3_bucket" "updates_bucket" {
+  bucket = "${local.bucket_name}-${random_id.rand_var.hex}"
+  acl    = "public-read"
+
+  cors_rule {
+    allowed_headers = ["*"]
+    allowed_methods = ["PUT", "POST"]
```

We'll only allow `GET` and `HEAD`.
```suggestion
    allowed_methods = ["GET", "HEAD"]
```